jquery.dynatable.js

```diff
@@ -967,6 +967,42 @@
         // force false boolean value to -1, true to 1, and tie to 0
         return comparison === false ? -1 : (comparison - 0);
       },
+      alphaNumeric: function(a, b, attr, direction) {
+        //swap if reverse
+        if(direction === -1) {
+          var temp = a;
+          a = b;
+          b = temp;
+        }
+
+        //extract display text
+        a = $(a[attr.toString()]).text();
+        b = $(b[attr.toString()]).text();
+
+        var aMatch, bMatch, a1, b1, i= 0, n,
+
+        regex =/(\.\d+)|(\d+(\.\d+)?)|([^\d.]+)|(\.\D+)|(\.$)/g;
+
+        //equal
+        if(a === b) return 0;
+
+        //match
+        aMatch = a.toLowerCase().match(regex);
+        bMatch = b.toLowerCase().match(regex);
+
+        //sort
+        while(i< aMatch.length){
+            if(!bMatch[i]) return 1;
+            a1= aMatch[i],
+            b1= bMatch[i++];
```

Comment:
Make this a little less confusing; we're doing two different things with this one line. We're first grabbing `bMatch[i]` and then incrementing `i` for the next loop iteration., so if `i` is 0, then we're setting `b1 = bMatch[0]` and then incrementing `i` to 1 in the same line which can throw people off who think that the return value of `i++` is the new value of i rather than the old.